### DISCUSSION TITLE
Configuring vim (question on documentation re: Vim Insert Mode remapping)

### DISCUSSION THREAD

**Discussion by RhetTbull** [BetterThanTomorrow/calva#1321] [Q&A]
I'm new to Calva and Clojure.  My workflow is VS Code and Vim key bindings....I'm very comfortable/productive using these two together and have zero interest in learning another editor or set of key bindings.  The experience so far has been almost enough to turn me off from learning clojure.  I followed the instructions in the [docs](https://calva.readthedocs.io/en/stable/vim.html) -- installed neovim, etc.  After a lot of frustration with key mappings, I think I have things almost working but I don't understand this part of the instructions on [Vim Insert Mode](https://calva.readthedocs.io/en/stable/vim.html#remap-vim-s-insert-mode):

```
Remap Vim’s Insert Mode

Remap vim’s insert mode keybinding to go into command mode by adding the following to your user settings:

"vim.insertModeKeyBindings": [
    {
        "before": ["j", "k"],
        "after": ["<esc>"]
    }
]
(Change before to whatever keybinding you are comfortable with!)
```

What does this do and why do I need to do it?  The other parts of the vim instructions explain why the config changes are needed but I can't figure this one out.  Thanks!

**Comment by PEZ** [comment]
Hello. I am not a Vim extension user so I might not correctly understand it. But I think it might be as simple as that Calva uses the `escape` key for dismissing inline results and that this can conflict with the default way to enter `vi` command mode. So this could be a suggestion to ”learn” some other key for command mode to avoid the conflict.

If you figure out what it is about you are welcome to send a PR updating the docs.

**Reply by RhetTbull** [reply]
Thanks -- there were other instructions in the vim docs regarding the escape key and I remapped Calva's use of escape.  But I don't understand the the issue with insert mode as I haven't found a conflict with Calva on that one and I'm not sure what the recommended remapping for insert mode actually does or why it's needed.

PS -- if I figure it out, I'll happily send a PR to avoid frustration for future users!
